src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/services/calendar/NextHearingDateIntervalReCalculator.java

```diff
@@ -0,0 +1,118 @@
+package uk.gov.hmcts.reform.wataskmanagementapi.services.calendar;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.reform.wataskmanagementapi.domain.entities.calendar.DateTypeIntervalData;
+import uk.gov.hmcts.reform.wataskmanagementapi.domain.entities.camunda.CamundaValue;
+import uk.gov.hmcts.reform.wataskmanagementapi.domain.entities.camunda.ConfigurationDmnEvaluationResponse;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+
+import static java.lang.Boolean.TRUE;
+import static uk.gov.hmcts.reform.wataskmanagementapi.domain.entities.calendar.NextHearingDateIntervalData.NEXT_HEARING_DATE_MUST_BE_WORKING_DAY_NEXT;
+import static uk.gov.hmcts.reform.wataskmanagementapi.services.calendar.DateType.NEXT_HEARING_DATE;
+
+@Slf4j
+@Component
+public class NextHearingDateIntervalReCalculator extends DueDateIntervalCalculator {
+
+    public NextHearingDateIntervalReCalculator(WorkingDayIndicator workingDayIndicator) {
+        super(workingDayIndicator);
+    }
+
+    @Override
+    public boolean supports(List<ConfigurationDmnEvaluationResponse> nextHearingDateProperties,
+                            DateType dateType,
+                            boolean isReconfigureRequest) {
+        ConfigurationDmnEvaluationResponse nextHearingDateOrigin = getReConfigurableProperty(
+            nextHearingDateProperties,
+            NEXT_HEARING_DATE_ORIGIN
+        );
+        ConfigurationDmnEvaluationResponse nextHearingDate = getReConfigurableProperty(
+            nextHearingDateProperties,
+            NEXT_HEARING_DATE.getType()
+        );
+        return NEXT_HEARING_DATE == dateType
+            && Optional.ofNullable(nextHearingDateOrigin).isPresent()
+            && Optional.ofNullable(nextHearingDate).isEmpty()
+            && isReconfigureRequest;
+    }
+
+    @Override
+    public ConfigurationDmnEvaluationResponse calculateDate(
+        List<ConfigurationDmnEvaluationResponse> nextHearingDateProperties,
+        DateType dateType) {
+        DateTypeIntervalData nextHearingDateIntervalData = readOriginFields(nextHearingDateProperties);
+
+        return calculateDate(dateType, nextHearingDateIntervalData);
+    }
+
+    private DateTypeIntervalData readOriginFields(List<ConfigurationDmnEvaluationResponse> dateTypeProperties) {
```

Comment:
Same comment as of due date re-calculator, you can use the method in super class with canReconfigure parameter